rasa/nlu/selectors/embedding_response_selector.py

```diff
@@ -0,0 +1,183 @@
+    and additional hidden layers are added together with dropout.
+    """
+
+    provides = ["response", "response_ranking"]
+
+    requires = ["text_features"]
+
+    name = "ResponseSelector"
+
+    # default properties (DOC MARKER - don't remove)
+    defaults = {
+        # nn architecture
+        # sizes of hidden layers before the embedding layer for input words
+        # the number of hidden layers is thus equal to the length of this list
+        "hidden_layers_sizes_a": [256, 128],
+        # sizes of hidden layers before the embedding layer for intent labels
+        # the number of hidden layers is thus equal to the length of this list
+        "hidden_layers_sizes_b": [256, 128],
+        # Whether to share the embedding between input words and intent labels
+        "share_hidden": False,
+        # training parameters
+        # initial and final batch sizes - batch size will be
+        # linearly increased for each epoch
+        "batch_size": [64, 256],
+        # how to create batches
+        "batch_strategy": "balanced",  # string 'sequence' or 'balanced'
+        # number of epochs
+        "epochs": 300,
+        # set random seed to any int to get reproducible results
+        "random_seed": None,
+        # embedding parameters
+        # dimension size of embedding vectors
+        "embed_dim": 20,
+        # the type of the similarity
+        "num_neg": 20,
+        # flag if minimize only maximum similarity over incorrect actions
+        "similarity_type": "auto",  # string 'auto' or 'cosine' or 'inner'
+        # the type of the loss function
+        "loss_type": "softmax",  # string 'softmax' or 'margin'
+        # how similar the algorithm should try
+        # to make embedding vectors for correct intent labels
+        "mu_pos": 0.8,  # should be 0.0 < ... < 1.0 for 'cosine'
+        # maximum negative similarity for incorrect intent labels
+        "mu_neg": -0.4,  # should be -1.0 < ... < 1.0 for 'cosine'
+        # flag: if true, only minimize the maximum similarity for
+        # incorrect intent labels
+        "use_max_sim_neg": True,
+        # scale loss inverse proportionally to confidence of correct prediction
+        "scale_loss": True,
+        # regularization parameters
+        # the scale of L2 regularization
+        "C2": 0.002,
+        # the scale of how critical the algorithm should be of minimizing the
+        # maximum similarity between embeddings of different intent labels
+        "C_emb": 0.8,
+        # dropout rate for rnn
+        "droprate": 0.2,
+        # visualization of accuracy
+        # how often to calculate training accuracy
+        "evaluate_every_num_epochs": 20,  # small values may hurt performance
+        # how many examples to use for calculation of training accuracy
+        "evaluate_on_num_examples": 0,  # large values may hurt performance,
+        # selector config
+        "response_type": None,
```

Comment:
what is this? or rather can we please come up with a better name?

Reply:
replacing it with `open_domain_intent_name`
It is exactly what the new name suggests - the open domain intent for which this response selector is to be trained. If it's left to None, a generic response selector will be trained which will collect examples from all open domain intents.

Reply:
I don't like `open_domain`, it psychologically limits the usage of this

Reply:
@Ghostvv any suggestions?

Reply:
`selector_type`?

Reply:
wait if it is actually intent, why don't we call it `corresponding_intent` or something?

Reply:
Good point. I like `corresponding_intent`

Reply:
I like the meaning, I don't like the word `corresponding`, do you have a synonym in mind?

Reply:
Not a synonym but another option - `end_to_end_intent_name`. Too long?

Reply:
yes 😅 too long

Reply:
why not just `intent`?

Reply:
because it's not any intent right? Only those intents which are participating in end to end response selection can be a part of it.

Reply:
I am going ahead with `direct_response_intent`. I am using the same in docs to distinguish intents needing end to end response from general intents.

Reply:
hm, @amn41 @tmbo any name suggestions?

Reply:
To be honest, `direct_response_intent` is not intuitive, I think simply `intent` is enough here, if it doesn’t contain responses, we’ll throw an error